marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs

```diff
@@ -70,6 +70,10 @@ function main(content, options) {
     datahub.debug.log({message: e, type: 'error'});
     throw Error(e);
   }
+  // fix the document URI if the format changes
+  if (content.uri && !content.uri.endsWith(outputFormat)) {
```

Comment:
Because this is duplicated, I recommend adding a function to a library that accepts a URI and an output format and if the "if" condition is met, a modified URI is returned; else, the original URI is returned. That function can then be covered via very small unit tests to address all the scenarios we can think of.

Reply:
Done. Moved it to a function in flowUtils and added unit test specifically for it.